### DISCUSSION TITLE
How to cache sources (not binary) ?

### DISCUSSION THREAD

**Discussion by marcovc** [microsoft/vcpkg#34227] [General]
When building the docker image of my app, vcpkg downloads and builds all dependencies. I am fine with the building part, the problem is the download part: if one website of some depency is down, then I can't no longer build my app. Is there a way to supply an offline copy of the dependencies source code, and instruct vcpkg to use that instead of downloading?

thanks!
Marco

**Comment by dg0yt** [comment]
- You can just save and restore the `downloads` directory.
- You can configure asset caching:
  https://learn.microsoft.com/vcpkg/users/assetcaching

**Reply by marcovc** [reply]
Restoring the downloads directory worked just fine, thanks!
